nautobot/core/views/__init__.py

```diff
@@ -284,29 +296,54 @@ def collect(self):
         yield gauge
 
 
-def nautobot_metrics_view(request):
-    """Exports /metrics.
+class PrometheusVersioning(URLPathVersioning):
+    """Overwrite the Nautobot API Version with the prometheus API version. Otherwise Telegraf/Prometheus won't be able to poll due to a version mismatch."""
```

Comment:
I'm curious - was this a problem previously? Were things working in a previous version and then broken along the way? Wondering in particular if this part of the fix will need back-porting to 1.6.

Reply:
Yes, the metrics endpoint is unauthenticated in 1.6.

I believe in the past, Prometheus didn't recommend putting authentication on the `/metrics` endpoint. I'm trying to find some references, but the guidance was basically "prometheus is for monitoring, put something else in front of it if you care about security".

There are really 2 things adding in the ability to require API keys prevents:
1. A publicly accessible Nautobot install from leaking metrics to the internet. Yes, you can/should protect it with nginx/traefik. The problem is, in some cases, Nautobot may be installed in Cloud A but my metrics stack lives in Cloud B. I may not necessarily know what IP address from Cloud B will scrape Nautobot with so I either have to allow all of Cloud B's prefixes or have scraping fail.
2. Apps exposing private data (sites, device names, ip addresses, etc.) to the internet. This [feature](https://docs.nautobot.com/projects/core/en/stable/development/apps/api/prometheus/) was added in 1.5.13 according to the docs. This would require Nautobot to be exposed to the internet AND using apps that add on to the `/metrics` endpoint

Reply:
Thanks for the info. I was actually asking specifically about the API versioning - can you comment on that too? :-)

Reply:
Previously, it wasn't an issue as we set it to `prometheus_client.CONTENT_TYPE_LATEST`. The issue is when using the DRF view, it sets it to Nautobot's API version. And then when Telegraf or Prometheus try to scrape, they get errors as they're looking for 0.0.4.